python/pypto/jit/decorator.py

```
@@ -639,10 +667,15 @@ def _compile(
-        source = Specializer(class_name, contexts, dynvar_bindings, dynvar_literals).specialize()
-        parsed = pl.parse(source)
-        skip_ptoas = not _ptoas_available()
-        return ir_compile(parsed, skip_ptoas=skip_ptoas, platform=platform)
+        specializer = Specializer(class_name, contexts, dynvar_bindings, dynvar_literals)
+        source = specializer.specialize()
+        rename_map = specializer.rename_map
+        try:
+            parsed = pl.parse(source)
+            skip_ptoas = not _ptoas_available()
+            return ir_compile(parsed, skip_ptoas=skip_ptoas, platform=platform)
+        except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)